### ISSUE TITLE
xhook breaks `navigator.sendBeacon(formData)` 

### ISSUE THREAD

**Issue by BenoitZugmeyer** [jpillora/xhook#151]
When using your library on a website, every usages of `navigator.sendBeacon(formData)` are broken.

It seems that your library is [patching the `FormData` object](https://github.com/jpillora/xhook/blob/main/src/patch/formdata.js). It works fine with `fetch` and `xhr` because it is also patching them. But it is still an issue with `sendBeacon`: the browser is receiving an unexpected [`Xhook`](https://github.com/jpillora/xhook/blob/main/src/patch/formdata.js#L13) instance and because it does not recognize it, it just run `toString` on the provided value which produces `"[object Object]"`.

### How to reproduce:

* With `xhook` installed on a HTML document, run something like this:
```js
const formData = new FormData()
formData.append("foo", "bar")
navigator.sendBeacon("/", formData)
```
* Look at the request payload in the devtools:
  * Actual:
![Screenshot 2022-07-29 at 17 43 51](https://user-images.githubusercontent.com/61787/181795807-7a2ac8e9-e341-45f1-be68-98aaf53677d1.png)
  * Expected: 
![Screenshot 2022-07-29 at 17 45 09](https://user-images.githubusercontent.com/61787/181795993-6ddbb38a-b219-4663-b1c8-e837afa21c58.png)

**Comment by jpillora** [comment]
Yea this is a tough one, definitely a bug. many years I needed to serialise/json a form data object and couldn’t find a way but maybe it’s possible now?

If we still can’t, we’ll need to patch sendBeacon too

**Comment by BenoitZugmeyer** [comment]
Yes you can inspect `FormData` now, on all major browsers except IE: https://developer.mozilla.org/en-US/docs/Web/API/FormData/entries

**Comment by jpillora** [comment]
Happy to remove patching `FormData`, and add with a `FormData` deserialise (into the current xhook API), pass it through all the hooks to be modified etc, then serialise back into `FormData` before firing off the request

@BenoitZugmeyer Feel free to send a PR, and please add a test for this, and make sure the other tests are passing

(How's this sound @TrickyPi ?)

**Comment by BenoitZugmeyer** [comment]
👋 I won't work on a fix because I don't use your library myself and don't have much time on my hands. I was just reporting the issue on behalf of a third party  (sorry!)

**Comment by TrickyPi** [comment]
No need to say sorry. Thanks for reporting this issue, we will fix it.
